build.clj

```diff
@@ -205,38 +228,74 @@
-                                (spit path (str js-resourced " ;;"))
-                                (spit path js-resourced))
-                              ctx)}])
-    (ra/start-figwheel!
-     {:figwheel-options (cond-> {:builds-to-start (if build-ids
-                                                    (->> (.split build-ids ",")
-                                                         (map (comp keyword string/trim))
-                                                         vec)
-                                                    [:android])}
-                          port      (merge {:nrepl-port       (some-> port Long/parseLong)
-                                            :nrepl-middleware ["cider.nrepl/cider-middleware"
-                                                               "refactor-nrepl.middleware/wrap-refactor"
-                                                               "cemerick.piggieback/wrap-cljs-repl"]}))
-      :all-builds       (into [] (for [[build-id {:keys [source-paths compiler warning-handlers]}]
-                                       (get-cljsbuild-config :dev)]
-                                   {:id               build-id
-                                    :source-paths     (conj source-paths "env/dev")
-                                    :compiler         compiler
-                                    :warning-handlers warning-handlers
-                                    :figwheel         true}))})
-    (rfs/-main)
-    (if-not port
-      (ra/cljs-repl)
-      (spit ".nrepl-port" port))))
+    (let [{:keys [port android-device ios-device]} (parse-figwheel-cli-opts args)
+          hosts-map                                {:android (resolve-dev-host :android android-device)
+                                                    :ios     (resolve-dev-host :ios ios-device)}
+          build-ids                                (if build-ids
+                                                     (->> (.split build-ids ",")
+                                                          (map (comp keyword str/trim))
+                                                          vec)
+                                                     [:android])]
+      (enable-source-maps)
+      (write-env-dev hosts-map)
+      (doseq [build-id build-ids
+              :let     [host-ip                 (get hosts-map build-id)
+                        platform-name           (if (= build-id :ios) "iOS" "Android")
+                        {:keys [js-modules
+                                app-name
+                                resource-dirs]} main-config]]
+        (rebuild-index-js build-id {:app-name      app-name
+                                    :host-ip       host-ip
+                                    :js-modules    js-modules
+                                    :resource-dirs resource-dirs})
+        (println-colorized (format "Dev server host for %s: %s" platform-name host-ip) green-color))
```

Comment:
I'm  having `Dev server host for Android: null` after running `clj -R:repl build.clj figwheel dev android`

Reply:
<img width="483" alt="screenshot 2018-05-09 15 58 07" src="https://user-images.githubusercontent.com/23836/39815841-fb820930-53a1-11e8-8039-67d598ec90b3.png">

`react-native run-android` gives "Network request failed" error.

`index.android.js` downloaded successfully.
` BUNDLE  [android, dev] ./index.android.js ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ 100.0% (878/878), done.`

Reply:
@dmitryn fixed, please try this now `clj -R:repl build.clj figwheel -p android`.

Reply:
@psdp thanks, it works on Android now.

Did you male build for ios? `react-native run-ios` gives me
```
status-react/node_modules/react-native/React/../third-party/glog-0.3.4/src/base/mutex.h:105:10: fatal error: 'config.h' file not found
#include "config.h"           // to figure out pthreads support
         ^~~~~~~~~~
1 error generated.


** BUILD FAILED **
```

Reply:
@psdp reinstalled everything from scratch, i'm getting this now:

```
** BUILD SUCCEEDED **


Installing build/Build/Products/Debug-iphonesimulator/StatusIm.app
An error was encountered processing the command (domain=NSCocoaErrorDomain, code=4101):
Couldn’t communicate with a helper application.
Try your operation again. If that fails, quit and relaunch the application and try again.
Launching im.status.ethereum.debug
im.status.ethereum.debug: -1
An error was encountered processing the command (domain=FBSOpenApplicationServiceErrorDomain, code=1):
The request to open "im.status.ethereum.debug" failed.
The request was denied by service delegate (SBMainWorkspace) for reason: NotFound ("Application "im.status.ethereum.debug" is unknown to FrontBoard").
Underlying error (domain=FBSOpenApplicationErrorDomain, code=4):
        The operation couldn’t be completed. Application "im.status.ethereum.debug" is unknown to FrontBoard.
```